### PR TITLE
Cigarettes heat their contents + early returns

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
- Cigarettes heat up their contents while they are lit.
- Candy cigarettes now caramelize their own sugar. (Their temperature has been reduced to prevent them from _charcoalizing_ it too.)
- Cleans up cigarette code a bit.

## Why It's Good For The Game
- More ways to boobytrap cigarettes than dipping them in welding fuel/plasma.
- Slightly improved cigarette code.

## Changelog
:cl:
add: Cigarettes heat their contents while they are lit.
balance: Candy cigarettes now caramelize their own sugar. Their temperature has been reduced to ensure they don't charcoalize it too. 
code: Cleaned up cigarette code a bit.
/:cl:

### CODE REVIEW

**Comment by norill** [comment]
>     * More ways to boobytrap cigarettes than dipping them in welding fuel/plasma.
> 


unlit cigs have no_react flag that gets cleared on lighting, which means you could easily rig them with lots of crazy stuff that would only go off when lit

**Comment by TemporalOroboros** [comment]
@norill 

The specific thing I was thinking of when I made this was life reaction blunts. It's quite a bit harder otherwise.

**Comment by norill** [comment]
> 
> 
> @norill
> 
> The specific thing I was thinking of when I made this was life reaction blunts. It's quite a bit more complex otherwise.

you can just heat Strange Reagent and Synthflesh in one beaker and Blood in another and then transfer everything to cigs with a dropper. add some oxygen and you can light them even in space

File: code/game/objects/items/cigs_lighters.dm [hunk 1]
```diff
@@ -20,16 +20,18 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	desc = "A simple match stick, used for lighting fine smokables."
 	icon = 'icons/obj/cigarettes.dmi'
 	icon_state = "match_unlit"
+	/// Whether this match has been lit.
 	var/lit = FALSE
+	/// Whether this match has burnt out.
 	var/burnt = FALSE
 	/// How long the match lasts in seconds
-	var/smoketime = 10
+	var/smoketime = 10 SECONDS
```
Comment (Fikou) [inline code/game/objects/items/cigs_lighters.dm:28]: new vars should be at the end

File: code/game/objects/items/cigs_lighters.dm [hunk 2]
```diff
@@ -20,16 +20,18 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
 	w_class = WEIGHT_CLASS_TINY
 	heat = 1000
 	grind_results = list(/datum/reagent/phosphorus = 2)
 
 /obj/item/match/process(delta_time)
-	smoketime -= delta_time
+	smoketime -= delta_time SECONDS
```
Comment (Mothblocks) [inline code/game/objects/items/cigs_lighters.dm:34]: `SECONDS` should really only be for constants, change this to `delta_time * 10` (or `delta_time * (1 SECONDS))`).

Comment (TemporalOroboros) [reply code/game/objects/items/cigs_lighters.dm:34]: Done.

File: code/game/objects/items/cigs_lighters.dm (hunk 1 above)
Comment (TemporalOroboros) [reply code/game/objects/items/cigs_lighters.dm]: Done.

**Pull request merged**